backend/src/controllers/userController.js

```diff
@@ -61,6 +61,19 @@ export default class UserController {
     });
   }
 
+// check if email already exists in database
+  checkEmail(email) {
+    return new Promise((resolve, reject) => {
+      const UserModel = new User();
+      UserModel.checkEmail(email, (err, result) => {
```

Comment:
essentially this UserModel.checkEmail part can be called in the signup, and we will check the returned result to see if it's empty or not and decide if we proceed

Reply:
Do we import User from the backend if I add the UserModel.checkEmail in signup
<img width="581" alt="Screenshot 2023-04-10 at 2 35 57 PM" src="https://user-images.githubusercontent.com/75223820/231003719-1461735d-f811-43a5-93af-aeb0dbb690bf.png">

Reply:
oh no you would check it all on the backend and you can return with a different 400 error message and the frontend checks for that. The frontend should never import from the backend like that

Reply:
I think I almost figured out a way but when I try to send an email query on Postman I get "message": "Unauthorized Error". 
<img width="550" alt="Screenshot 2023-04-14 at 6 30 49 PM" src="https://user-images.githubusercontent.com/75223820/232176719-5186a5ee-d88f-4d90-8a3d-35627b45d9e3.png">
<img width="442" alt="Screenshot 2023-04-14 at 6 40 18 PM" src="https://user-images.githubusercontent.com/75223820/232177056-69e2fc2a-9f43-44b7-a8ea-53ca5e29c5f3.png">

Reply:
It's possible that the fetch user call your making requires you to be auth'd, I would check that first.